src/sync/app_credentials.hpp

```diff
@@ -64,10 +66,16 @@ struct AppCredentials {
     // Construct and return anonymous credentials
     static AppCredentials anonymous();
 
-     // Construct and return credentials from an Apple account token.
+    // Construct and return credentials from an Apple account token.
     static AppCredentials apple(const AppCredentialsToken id_token);
-
-     // Construct and return credentials from a username and password.
+    
+    // Construct and return credentials from a google account token.
+    static AppCredentials google(const AppCredentialsToken id_token);
```

Comment:
All of these seem to lack tests? How do we know they actually work?

Reply:
I've added unit tests to make sure each auth provider produces the correct json